crates/cli/src/subcommands/generate/mod.rs

```diff
@@ -240,114 +245,132 @@ pub struct GenCtx {
     names: Vec<Option<String>>,
 }
 
-pub fn generate(module: RawModuleDefV8, lang: Language, namespace: &str) -> anyhow::Result<Vec<(String, String)>> {
-    let (ctx, items) = extract_from_moduledef(module);
-    let items: Vec<GenItem> = items.collect();
-    let mut files: Vec<(String, String)> = items
-        .iter()
-        .filter_map(|item| item.generate(&ctx, lang, namespace))
-        .collect();
-    files.extend(generate_globals(&ctx, lang, namespace, &items));
-
-    Ok(files)
+pub fn generate(module: RawModuleDef, lang: Language, namespace: &str) -> anyhow::Result<Vec<(String, String)>> {
+    let module = match module {
```

Comment:
Also this whole function looks good, this was exactly the structure I thought would make sense here while we port